packages/protobuf/src/from-binary.ts

```diff
@@ -108,12 +108,12 @@ function readMessage(
   const unknownFields = message.getUnknown() ?? [];
   while (reader.pos < end) {
     [fieldNo, wireType] = reader.tag();
-    if (wireType == WireType.EndGroup) {
+    if (delimited && wireType == WireType.EndGroup) {
```

Comment:
We should only break the loop for EndGroup if we're currently parsing a group (a.k.a. delimited message encoding).

Previously, this would behave incorrectly for a rogue EndGroup tag.